app/assets/javascripts/modules/single-file-upload.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+moj.Modules.singleFileUpload = {
+  input_class: '.single-file-upload',
+  spinnerImagePath: '/assets/spinner.gif',
+
+  init: function() {
+    var self = this;
+
+    if($(self.input_class).length) {
+      self.bindEvents();
+      self.preloadSpinner();
+    }
+  },
+
+  bindEvents: function() {
+    var self = this,
+        $el = $(self.input_class).eq(0),
+        $form = $el.closest('form');
+
+    $el.on('change', function() {
+      var statusText = '';
+
+      if($el.val()) {
+        statusText = 'SELECTED';
```

Comment:
Ah, yes. Good point. We can make this a `.js.erb` file, no problem. How much hassle did caching cause? Because this definitely sounds worth doing to me.